### PR TITLE
chore: Remove position props from List and Table widgets

### LINKED ISSUE #26629: [Task]: Cleanup dataTreeFactory to avoid exposing position properties
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

dataTreeFactory is adding `appPositioningType` and `isMobile` properties to the DataTree. This is used to derive height of the widgets within some widget derived properties.
This should be avoided and we should be directly providing height & width of the widget into the DataTree.

### LINKED ISSUE #26981: [Task]: Cleanup ListWidget to avoid usage of position properties
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

There are multiple usage of position properties like topRow, BottomRow, rightColumn, leftColumn within list widget's code base as well as within MetaWidgetGenerator.
List should be using layout independent abstractions instead.

### PR DESCRIPTION
**Pull request by aswathkk**
## Description
- Replaced the property `templateBottomRow` with `templateHeight` as a first step to remove the dependency of fixed layout specific position property from List widget.
- Remove appPositioningType and isMobile props from widget DataTree.
- Include width and height as `componentHeight` and `componentWidth` in widget DataTree.
- Update ListWidget, ListWidgetV2, TableWidget and TableWidgetV2 to use the componentHeight prop in derivedProps calculation to get pageSize.

Next set of task would be done as a part of https://github.com/appsmithorg/appsmith/issues/26982 and https://github.com/appsmithorg/appsmith/issues/27050

#### PR fixes following issue(s)
Fixes #26981 
Fixes #26629

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/selectors/widgetSelectors.ts [hunk 1]
```diff
@@ -228,3 +238,49 @@ export const isResizingOrDragging = createSelector(
...
+  flattenedChildCanvasWidgets: Record<string, FlattenedWidgetProps> = {},
+) {
+  const parentWidget = canvasWidgets[parentWidgetId];
+  const { propertyEnhancer } = getLayoutSystem(renderMode, appPositioningType);
+  parentWidget?.children?.forEach((childId) => {
+    const childWidget = canvasWidgets[childId];
+    flattenedChildCanvasWidgets[childId] = propertyEnhancer({
+      ...childWidget,
+      isMobile,
+      parentRowSpace: childWidget.parentRowSpace
```
Comment (rahulramesha) [inline app/client/src/selectors/widgetSelectors.ts:257]: Could you please recheck this conditional logic? if it is correct could you make it more readable..

**Review by rahulramesha — CHANGES_REQUESTED** [review]

**Review by rahulramesha — DISMISSED** [review]
@aswathkk looks good to me, please also get a review from List widget code owner for the list widget changes.

**Review by sbalaji1192 — APPROVED** [review]

**Review by rahulramesha — DISMISSED** [review]

File: app/client/src/layoutSystems/common/utils/ComponentSizeUtils.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+}: BaseWidgetProps) => {
+  return {
+    componentWidth: (rightColumn - leftColumn) * parentColumnSpace,
+    componentHeight:
+      (bottomRow - topRow) *
+      (parentRowSpace || GridDefaults.DEFAULT_GRID_ROW_HEIGHT),
+  };
+};
+
+export const getComponentDimensions = memoize(
```
Comment (sbalaji1192) [inline app/client/src/layoutSystems/common/utils/ComponentSizeUtils.ts:80]: nit: We can leverage the experimental decorator to implement the memorize as it's more readable.

Comment (sbalaji1192) [reply app/client/src/layoutSystems/common/utils/ComponentSizeUtils.ts:80]: By default, lodash memo only uses the first argument as the cache, you should use a resolver function to have more than one argument.
lets use `micro-memoize` across the board.

Comment (aswathkk) [reply app/client/src/layoutSystems/common/utils/ComponentSizeUtils.ts:80]: decorators can be used only in the case of class methods. So, shall I use the mirco-memoize directly?

Comment (sbalaji1192) [reply app/client/src/layoutSystems/common/utils/ComponentSizeUtils.ts]: Yeah. we can make these functions members of a class. But Feel free to ignore it if you think that would be too much

Comment (sbalaji1192) [reply app/client/src/layoutSystems/common/utils/ComponentSizeUtils.ts]: Yeah. let's use micro-memoize.

**Review by sbalaji1192 — APPROVED** [review]

**Pull request merged**